File: app/src/main/java/br/com/edrsantos/santandertesteandroidv2/adapter/StatementsRecyclerViewAdapter.java
package br.com.edrsantos.santandertesteandroidv2.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.text.DecimalFormatSymbols;
import java.text.NumberFormat;
import java.util.List;
import java.util.Locale;

import br.com.edrsantos.santandertesteandroidv2.R;
import br.com.edrsantos.santandertesteandroidv2.model.Statement;
import br.com.edrsantos.santandertesteandroidv2.util.Util;

public class StatementsRecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    List<Statement> statementList;
    Context mContext;
    NumberFormat decimalFormat = new DecimalFormat("#,##0.00", new DecimalFormatSymbols(new Locale ("pt", "BR")));

    public StatementsRecyclerViewAdapter(List<Statement> statementList, Context mContext){
        this.statementList = statementList;
        this.mContext = mContext;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerView.ViewHolder viewHolder = null;
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        RecyclerView.LayoutParams layoutParams = new RecyclerView.LayoutParams(RecyclerView.LayoutParams.MATCH_PARENT, RecyclerView.LayoutParams.WRAP_CONTENT);
        View v = inflater.inflate(R.layout.item_statement, null, false);
        v.setLayoutParams(layoutParams);
        viewHolder = new MyViewHolder(v);

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        final MyViewHolder viewHolder = (MyViewHolder) holder;

        Statement s = statementList.get(position);

        viewHolder.title.setText(s.getTitle());
        viewHolder.desc.setText(s.getDesc());
        viewHolder.date.setText(Util.convertDateToString(s.getDate()));

        String value = decimalFormat.format(s.getValue());
        viewHolder.value.setText(value);
    }

    @Override
    public int getItemCount() {
        return statementList.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView title;
        TextView desc;
        TextView date;
        TextView value;

        public MyViewHolder(View itemView) {
            super(itemView);

            title = (TextView) itemView.findViewById(R.id.title);
            desc = (TextView) itemView.findViewById(R.id.desc);
            date = (TextView) itemView.findViewById(R.id.date);
            value = (TextView) itemView.findViewById(R.id.value);

        }
    }

}

File: app/src/main/java/br/com/cauejannini/testesantander/statements/StatementsRecyclerViewAdapter.kt
package br.com.cauejannini.testesantander.statements

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import br.com.cauejannini.testesantander.R
import br.com.cauejannini.testesantander.commons.Utils
import kotlinx.android.synthetic.main.list_item_statement.view.*

class StatementsRecyclerViewAdapter(val statementList: List<Statement>):
    RecyclerView.Adapter<StatementsRecyclerViewAdapter.StatementItemViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatementItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item_statement, parent, false)
        return StatementItemViewHolder(view)
    }

    override fun getItemCount(): Int {
        return statementList.size
    }

    override fun onBindViewHolder(holder: StatementItemViewHolder, position: Int) {
        val statement = statementList.get(position)
        holder.tvTitle.text = statement.title
        holder.tvDate.text = Utils.dataApiParaApp(statement.date)
        holder.tvDesc.text = statement.desc

        val value = Utils.doisDecimais(statement.value)
        holder.tvValue.text = "R$ $value"
    }

    class StatementItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        val tvTitle = itemView.tvTitle
        val tvDesc = itemView.tvDesc
        val tvDate = itemView.tvDate
        val tvValue = itemView.tvValue

    }

}